.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -11,107 +11,197 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+# Creates the owl-bot binary (no node runtime needed)
 
-# install gapic-generator-java in a separate layer so we don't overload the image
-# with the transferred source code and jars
-FROM gcr.io/cloud-devrel-public-resources/java21 AS ggj-build
+# node:22.1-alpine
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/node@sha256:487dc5d5122d578e13f2231aa4ac0f63068becd921099c4c677c850df93bede8 as owlbot-cli-build
+ARG OWLBOT_CLI_COMMITTISH=ac84fa5c423a0069bbce3d2d869c9730c8fdf550
+
+# install tools
+RUN apk add git
+
+# Clone the owlbot-cli source code
+WORKDIR /tools
+RUN git clone https://github.com/googleapis/repo-automation-bots
+WORKDIR /tools/repo-automation-bots/packages/owl-bot
+RUN git checkout "${OWLBOT_CLI_COMMITTISH}"
+
+# Part of the code path (that we don't use) ends up touching a dependency called
+# @google-cloud/datastore that tries a fs.readFileSync that is not handled by
+# default by esbundle (esbundle is good a figuring out imports but doesn't
+# actively scan filesystem interactions such as fs.readFileSync). This makes the
+# app to fetch a file at runtime that is not available in the bundle context.
+# This is why we remove this import and its usage from the entrypoint.
+RUN sed -i '/testWebhook/d' src/bin/owl-bot.ts
+
+# Bundle the source code and its dependencies into a single javascript file
+# with all its dependencies embedded.
+# This is because SEA (see below) cannot
+# resolve external modules in a multi-file project.
+# We use the esbuild tool. See https://esbuild.github.io/
+COPY ./.cloudbuild/library_generation/image-configuration/owlbot-cli-build-config.js .
+RUN npm i esbuild
+RUN node owlbot-cli-build-config.js
```

Comment:
In order to create a SEA, do we have to bundle source code into a single js file?

Reply:
Yes, bundling is necessary:
https://github.com/googleapis/sdk-platform-java/blob/fb98222ddd61aa5b0e5e061dd8e4ffcbc12fdea2/.cloudbuild/library_generation/library_generation.Dockerfile#L39-L40

From the [docs](https://nodejs.org/api/single-executable-applications.html): _The single executable application feature currently only supports running a single embedded script using the CommonJS module system._

> Can be combine the two steps?
Do you mean using a single `RUN` + `&&`?

Reply:
I was referring whether we can build a SEA from multiple js source code, looks like it has to be bundled together first.